tools/block-generator/generator/generate.go

```diff
@@ -759,6 +765,93 @@ func (g *generator) generateAssetTxnInternalHint(txType TxTypeID, round uint64,
 	return
 }
 
+// ---- 3. App Transactions ----
+
+func (g *generator) generateAppTxn(round uint64, intra uint64) (transactions.SignedTxn, transactions.ApplyData, error) {
+	start := time.Now()
+	selection, err := weightedSelection(g.appTxWeights, getAppTxOptions(), appSwapCall)
+	if err != nil {
+		return transactions.SignedTxn{}, transactions.ApplyData{}, err
+	}
+
+	actual, txn, err := g.generateAppCallInternal(selection.(TxTypeID), round, intra, 0, nil)
+	if err != nil {
+		return transactions.SignedTxn{}, transactions.ApplyData{}, fmt.Errorf("unexpected error received from generateAppCallInternal(): %w", err)
+	}
+	if txn.Type == "" {
+		return transactions.SignedTxn{}, transactions.ApplyData{}, fmt.Errorf("missing transaction type for app transaction")
+	}
+	defer g.recordData(actual, start)
+
+	return signTxn(txn), transactions.ApplyData{}, nil
+}
+
+func (g *generator) generateAppCallInternal(txType TxTypeID, round, intra, hintIndex uint64, hintApp *appData) (TxTypeID, transactions.Transaction, error) {
```

Comment:
hintIndex not used.

Reply:
I'd like to keep this param until my follow up PR is ready for review, at which point I commit to removing if still unused.

Let's not resolve this discussion.

Reply:
In fact, I did end up using the hint in the next PR.

Reply:
it's removed in #5478